code/obj/machinery/traymachines.dm

```diff
@@ -0,0 +1,700 @@
+				light.set_color(0, 0, 0)
+				light.set_brightness(0)
+
+
+//-----------------------------------------------------
+/*~ Tanning Computer ~*/
+//-----------------------------------------------------
+
+/obj/machinery/computer/tanning
+	name = "tanning computer"
+	desc = "Used to control a tanning bed."
+	icon = 'icons/obj/stationobjs.dmi'
+	mats = 20
+	id = 2
+	icon_state = "tanconsole"
+	var/state_str = ""
+	var/obj/machinery/traymachine/locking/tanning/linked = null //The linked tanning bed
+
+	New()
+		..()
+		get_link()
+
+	disposing()
+		src.linked?.linked = null
+		src.linked = null
+		. = ..()
+
+	proc/get_link()
+		for(var/obj/machinery/traymachine/locking/tanning/C in by_type[/obj/machinery/traymachine/locking/tanning])
+			if(C.z == src.z && C.id == src.id && C != src)
+				linked = C
+				C.linked = src
+				break
+
+	proc/find_tray_tube()
+		if (linked.my_tray && istype(linked.my_tray, /obj/machine_tray/tanning))
+			var/obj/machine_tray/tanning/tray = linked.my_tray
+			if (tray.tanningtube)
+				return 1
+
+	proc/get_state_string()
+		if(linked == null) get_link()
+		if(linked == null) return "ERROR: No tanning beds found."
+
+		if(find_tray_tube() != 1) return "No light tube found in the tanning tray."
+		if(linked.locked) return "Tanning in progress. Please wait."
+		if(linked.locked == 0) return "Tanning bed idle."
+
+		return "Unknown Error Encountered."
+
+	attack_hand(var/mob/user as mob, params)
+		if (..(user))
+			return
+
+		state_str = src.get_state_string()
+
+		var/dat = ""
+		dat += "<b>Tanning Bed Status:</b><BR>"
+		dat += "[state_str]<BR>"
+		dat += "Set Time: [linked ? linked.settime : "--"]<BR>"
+		dat += "<b>Tanning Bed Control:</b><BR>"
+		dat += "<A href='?src=\ref[src];toggle=1'>Activate Tanning Bed</A><BR>"
+		dat += "<A href='?src=\ref[src];timer=1'>Delayed Activation</A><BR>"
+		dat += "<A href='?src=\ref[src];settime=1'>Increase Time</A><BR>"
+		dat += "<A href='?src=\ref[src];unsettime=1'>Decrease Time</A><BR>"
+
+		if (user.client.tooltipHolder)
+			user.client.tooltipHolder.showClickTip(src, list(
+				"params" = params,
+				"title" = src.name,
+				"content" = dat,
+			))
+
+		return
+
+	Topic(href, href_list)
+		if (..(href, href_list))
+			return
+
+		if (href_list["toggle"])
+			if (linked && !linked.locked && find_tray_tube() == 1 && linked.my_tray.loc == linked)
+				playsound(src.loc, "sound/machines/bweep.ogg", 20, 1)
+				linked.cremate()
+				logTheThing("station", usr, null, "activated the tanning bed at [usr.loc.loc] ([showCoords(usr.x, usr.y, usr.z)])")
+
+		else if (href_list["timer"])
+			sleep (10 SECONDS)
+			if (linked && !linked.locked && find_tray_tube() == 1 && linked.my_tray.loc == linked)
+				playsound(src.loc, "sound/machines/bweep.ogg", 20, 1)
+				linked.cremate()
+				logTheThing("station", usr, null, "activated the tanning bed at [usr.loc.loc] ([showCoords(usr.x, usr.y, usr.z)])")
+
+		else if (href_list["settime"])
+			if (linked && linked.settime < 20)
```

Comment:
Time defines

Reply:
I put the define already in the tanning bed's proc (line 497 atm), is there a reason to use them here?

(also because the UI is a bit stinky and you have to click for each second, adding the defines here would probably aggravate that by a factor 10 :P)

Reply:
I see an argument either way, whichever's fine, personally

Reply:
> I put the define already in the tanning bed's proc (line 497 atm), is there a reason to use them here?

what is 20 though